### ISSUE TITLE
Risc-v cannot compile without RVV

### ISSUE THREAD

**Issue by k0tran** [google/highway#2311]
Attempt to compile on a riscv64 target with `-DHWY_CMAKE_RVV=OFF` gave this error:
```log
In file included from /usr/src/RPM/BUILD/highway-1.2.0/hwy/ops/rvv-inl.h:19,
                 from /usr/src/RPM/BUILD/highway-1.2.0/hwy/highway.h:596,
                 from /usr/src/RPM/BUILD/highway-1.2.0/hwy/per_target.cc:28,
                 from /usr/src/RPM/BUILD/highway-1.2.0/hwy/foreach_target.h:314,
                 from /usr/src/RPM/BUILD/highway-1.2.0/hwy/per_target.cc:27:
/usr/lib64/gcc/riscv64-alt-linux/13/include/riscv_vector.h:32:2: error: #error "Vector intrinsics require the vector extension." 
   32 | #error "Vector intrinsics require the vector extension." 
```

This error is strange, considering RVV was explicitly turned off.

Temporary fix I found:
Add `-DHWY_COMPILE_ONLY_SCALAR` to compiler options

So in the end is this proper behaviour? Maybe HWY_CMAKE _RVV should set some additional flags for highway?

**Comment by jan-wassenberg** [comment]
Interesting, this is unexpected.
Can you confirm `HWY_BASELINE_RVV == 0 && HWY_HAVE_RUNTIME_DISPATCH_RVV == 0`?

**Comment by k0tran** [comment]
Yes, that's right. With `-DHWY_CMAKE_RVV` I got "RVV defines detected" after adding these lines at the start of the highway/hwy/ops/rvv-inl.h file:

```
#if HWY_BASELINE_RVV == 0
#if HWY_HAVE_RUNTIME_DISPATCH == 0
#error "RVV defines detected"
#else
#error "HWY_HAVE_RUNTIME_DISPATCH != 0"
#endif
#else
#error "HWY_BASELINE_RVV != 0"
#endif
```

**Comment by jan-wassenberg** [comment]
Thanks for the data, I am trying to trace what happened.
Without `HWY_TARGET == HWY_RVV`, we will not include `rvv-inl.h` where the error occurs.
`detect_targets.h` decides the bitfield of all `HWY_TARGET`. `HWY_RVV` only occurs in two places: `HWY_BASELINE_RVV`, which you verified is 0, and `HWY_ATTAINABLE_RISCV`. Thus it should be the latter, which happens if `HWY_ARCH_RISCV && HWY_HAVE_RUNTIME_DISPATCH`. But you also verified that `HWY_HAVE_RUNTIME_DISPATCH` (not the `HWY_HAVE_RUNTIME_DISPATCH_RVV` I mentioned) is zero.

You did mention that `HWY_COMPILE_ONLY_SCALAR` helps. So let's look at the default alternative to that: 1) and 1b) cannot set `HWY_RVV`. `HWY_RVV` is likely coming from `HWY_STATIC_TARGET` - let's confirm that `HWY_STATIC_TARGET == HWY_RVV`?

But `HWY_STATIC_TARGET` is the lowest set bit in `HWY_BASELINE_TARGETS`, and that is set via `HWY_BASELINE_RVV`, which was 0. So I do not understand how this can be happening. One more idea would be to modify `detect_targets.h` to put an `#error` inside the two `#if` where `HWY_RVV` is used: inside `#if HWY_ARCH_RISCV && HWY_HAVE_RUNTIME_DISPATCH` and `#if HWY_ARCH_RISCV && defined(__riscv_v_intrinsic) && \
    __riscv_v_intrinsic >= 11000`. Do those trigger?

**Comment by k0tran** [comment]
Error within `HWY_HAVE_RUNTIME_DISPATCH` is triggered. The one within `__riscv_v_intrinsic` condition is not

**Comment by jan-wassenberg** [comment]
Makes sense, thanks. So `HWY_ARCH_RISCV && HWY_HAVE_RUNTIME_DISPATCH`. I think this contradicts the previous finding that `HWY_HAVE_RUNTIME_DISPATCH == 0`?
So `HWY_HAVE_RUNTIME_DISPATCH_RVV != 0`, which is just `HWY_ARCH_RISCV && HWY_COMPILER_CLANG >= 1900`.
We'll disable the experimental `HWY_HAVE_RUNTIME_DISPATCH_RVV` again, Clang isn't ready for it. That should fix this.
